deployments/docker/Dockerfile

```diff
@@ -0,0 +1,136 @@
+# Args needs to be renewed for each build stage that uses these args
+ARG workDir=/app
+
+FROM golang:latest AS build-base
+
+# Renew Arg and set working directory to /app
+ARG workDir
+WORKDIR $workDir
+
+# Copy go installation files
+COPY ./go.mod ./
+COPY ./go.sum ./
+
+# Install build dependencies and tools
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install \
+        --assume-yes \
+        --no-install-recommends \
+        apt-transport-https \
+        apt-utils \
+        jq \
+        tree \
+        protobuf-compiler \
+        python3 \
+        python3-pip \
+        libprotobuf-dev \
+        libsqlite3-dev \
+        sqlite3 \
+        zlib1g-dev \
+    && apt-get clean \
+    && apt-get autoremove --assume-yes \
+    && rm -rf /var/lib/apt/lists/* /var/tmp/* /tmp/*
+
+# Install config templating dependencies
+RUN pip install Jinja2
+
+# Download Go modules
+RUN go mod download &&\
+    go install github.com/golang/mock/mockgen@v1.6.0 &&\
+    go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.26 &&\
+    go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1 &&\
+    go install github.com/mitchellh/protoc-gen-go-json@v1.1.0 &&\
+    go install github.com/veraison/corim/cocli@demo-psa-1.0.0 &&\
+    go install github.com/veraison/evcli@demo-psa-1.0.0
+
+
+FROM build-base as common-build
+# Renew build arguments
+ARG workDir
+ARG PROVISIONING_DEPLOY_PREFIX
+ARG VERIFICATION_DEPLOY_PREFIX
+ARG VTS_DEPLOY_PREFIX
+ARG VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS
+ARG VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS
+ARG VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS_PORT
+ARG VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS_PORT
+ARG BIN_DIR
+ARG LOG_DIR
+ARG PLUGIN_DIR
+ARG INPUT_FILE_DIR
+
+# Set envrionment variables for deploy directories and communication ip addresses
+ENV PROVISIONING_DEPLOY_PREFIX $workDir$PROVISIONING_DEPLOY_PREFIX
+ENV VERIFICATION_DEPLOY_PREFIX $workDir$VERIFICATION_DEPLOY_PREFIX
+ENV VTS_DEPLOY_PREFIX $workDir$VTS_DEPLOY_PREFIX
+ENV VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS $VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS
+ENV VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS $VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS 
+ENV VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS_PORT $VTS_PROVISIONING_COMMUNICATION_IP_ADDRESS_PORT
+ENV VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS_PORT $VTS_VERIFICATION_COMMUNICATION_IP_ADDRESS_PORT
+
+# Set environment variables for binaries, logs, plugins and input directories
+ENV BIN_DIR $BIN_DIR
+ENV LOG_DIR $LOG_DIR
+ENV PLUGIN_DIR $PLUGIN_DIR
+ENV INPUT_FILE_DIR $INPUT_FILE_DIR
+
+# Copy services into the image (TODO: git clone)
```

Comment:
how are we cloning the code at the moment? I think, git  clone is essential?

Reply:
The user always has to clone the veraison/services repo locally, in order to have visibility of the docker files to run `docker compose` and currently docker simply copies the local copy of the repo into the image. Git clone may be a cleaner way to get the code inside container, Ill look into how this can be done.

Reply:
Excellent ! Thank you!